crates/gen/src/com_interface.rs

```diff
@@ -61,24 +85,16 @@ impl ComInterface {
                 TokenStream::new()
             };
 
-            let params = method.signature.params.iter().map(|t| {
-                let name = format_ident(&t.name);
-                let tokens = t.gen_field();
-                quote! { #name: #tokens }
-            });
-
-            let args = method.signature.params.iter().map(|t| {
-                let name = format_ident(&t.name);
-                quote! { #name }
-            });
-
+            let constraints = gen_constraint(method);
+            let params = gen_params(method);
+            let args = gen_abi_args(method);
             let name = method.gen_name();
             let vtable_offset = Literal::u32_unsuffixed((vtable_offset + 3) as u32);
 
             quote! {
-                pub fn #name(&self, #(#params),*) #return_type {
+                pub fn #name<#constraints>(&self, #params) #return_type {
```

Comment:
nit: I know it was like this before, but it would be nicer if `return_type` really was just the type and not also the `->`. This might make it easier to understand and compose.

Reply:
Good idea, but I'll leave that as homework - this pattern appears in a few places and I'd like to tackle them all if possible.